pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -283,39 +231,38 @@ func (c *Cluster) reconcileServices(activeDaemon string) error {
 
 	// enable monitoring if `monitoring: enabled: true`
 	if c.spec.Monitoring.Enabled {
-		if err := c.EnableServiceMonitor(activeDaemon); err != nil {
+		if err := c.EnableServiceMonitor(); err != nil {
 			return errors.Wrap(err, "failed to enable service monitor")
 		}
 	}
 
-	return c.updateServiceSelectors(activeDaemon)
+	return c.updateServiceSelectors()
 }
 
-// Make a best effort to update the services that have been labeled for being updated
-// when the mgr has changed. They might be services for node ports, ingress, etc
-func (c *Cluster) updateServiceSelectors(activeDaemon string) error {
+// For the upgrade scenario: remove any selector DaemonIDLabel from the all
+// the services since new mgr HA doesn't rely on this label anymore.
+func (c *Cluster) updateServiceSelectors() error {
```

Comment:
no need to return an error
```suggestion
func (c *Cluster) updateServiceSelectors() {
```

Reply:
this error is propagated in other function calls so I'm afraid we cannot avoid returning it.

Reply:
I only see a couple of simple calls to this method. Since this method only ever returns `nil` (and never an actual error), and it's a private method, let's go ahead and change it.

Reply:
Fixed